src/Compilers/VisualBasic/Portable/Semantics/TypeInference/TypeArgumentInference.vb

```diff
@@ -1633,6 +1629,10 @@ HandleAsAGeneralExpression:
                 inferenceRestrictions As RequiredConversion
             ) As Boolean
 
+                If Not RefersToGenericParameterToInferArgumentFor(parameterType) Then
```

Comment:
>RefersToGenericParameterToInferArgumentFor [](start = 23, length = 42)

I think the name of `RefersToGenericParameterToInferArgumentFor` should be `RefersToGenericParameterToInferTypeArgument` if I understand its purpose.